compiler/rustc_mir_build/src/thir/pattern/mod.rs

```diff
@@ -93,6 +93,10 @@ impl<'a, 'tcx> PatCtxt<'a, 'tcx> {
             None => Ok((None, None)),
             Some(expr) => {
                 let (kind, ascr) = match self.lower_lit(expr) {
+                    PatKind::InlineConstant { subpattern, value } => (
+                        PatKind::Constant { value: Const::Unevaluated(value, subpattern.ty) },
+                        None,
+                    ),
                     PatKind::AscribeUserType { ascription, subpattern: box Pat { kind, .. } } => {
                         (kind, Some(ascription))
                     }
```

Comment:
Could there be an inline constant with a type ascription around it? I'm not sure how to get a type ascription at all in a range pattern actually, `(0:i8)..=10` and `{0:i8}..=10` don't parse

Reply:
Also in this function `subpattern` must already be `PatKind::Constant`, may as well return it directly instead, so that the only code that touches `value` is unsafety checking

Reply:
Oh, I see you rely on the `Const::Unevaluated` somewhere else. This feels brittle to me, I would like to be able to evaluate constants freely (is that a mistaken assumption?).

Could we do like the type ascriptions and re-wrap the outputted range pattern with the `InlineConstant`? It would change the meaning of `InlineConstant` slightly, instead of "this pattern came from this inline constant", it would mean "this pattern contained this inline constant somewhere". I'm asking because I was working on making `PatRange` carry only `u128`s since we keep `eval_bits`ing the endpoints.

Reply:
I can do that. It does make the meaning of things a bit unintuitive, but simplify the consuming code.